### PR TITLE
community: add include_labels option to ConfluenceLoader

### PR DESCRIPTION
**Pull request by nakamasato**
## **Description:**

Enable `ConfluenceLoader` to include labels with `include_labels` option (`false` by default for backward compatibility). and the labels are set to `metadata` in the `Document`. e.g. `{"labels": ["l1", "l2"]}`

## Notes

Confluence API supports to get labels by providing `metadata.labels` to `expand` query parameter 

All of the following functions support `expand` in the same way:
- confluence.get_page_by_id
- confluence.get_all_pages_by_label
- confluence.get_all_pages_from_space
- cql (internally using [/api/content/search](https://developer.atlassian.com/cloud/confluence/rest/v1/api-group-content/#api-wiki-rest-api-content-search-get))

## **Issue:**

No issue related to this PR.

## **Dependencies:** 

No changes.

## **Twitter handle:** 

[@gymnstcs](https://x.com/gymnstcs)

- [x] **Add tests and docs**: If you're adding a new integration, please include
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.

- [x] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

### CODE REVIEW

File: libs/community/langchain_community/document_loaders/confluence.py [hunk 1]
```diff
@@ -575,10 +592,19 @@ def process_page(
...
+                for label in page.get("metadata", {})
+                .get("labels", {})
+                .get("results", [])
+            ]
+
         metadata = {
             "title": page["title"],
             "id": page["id"],
             "source": self.base_url.strip("/") + page["_links"]["webui"],
+            **({"labels": labels} if include_labels else {}),
```
Comment (nakamasato) [inline libs/community/langchain_community/document_loaders/confluence.py:607]: set `labels` key only when `include_labels` is set to true

File: libs/community/langchain_community/document_loaders/confluence.py [hunk 2]
```diff
@@ -575,10 +592,19 @@ def process_page(
             ]
             text = text + "".join(comment_texts)
 
+        if include_labels:
+            labels = [
+                label["name"]
+                for label in page.get("metadata", {})
+                .get("labels", {})
+                .get("results", [])
+            ]
```
Comment (nakamasato) [inline libs/community/langchain_community/document_loaders/confluence.py:601]: label would be sth like this: `{'prefix': 'global', 'name': 'database', 'id': '111111111'}` 

Screenshot 2024-11-21 at 22 23 03

[ref](https://developer.atlassian.com/cloud/confluence/rest/v1/api-group-content/#api-wiki-rest-api-content-search-get)

File: libs/community/langchain_community/document_loaders/confluence.py [hunk 3]
```diff
@@ -327,12 +329,20 @@ def _lazy_load(self, **kwargs: Any) -> Iterator[Document]:
...
         ocr_languages = self._resolve_param("ocr_languages", kwargs)
         keep_markdown_format = self._resolve_param("keep_markdown_format", kwargs)
         keep_newlines = self._resolve_param("keep_newlines", kwargs)
+        expand = ",".join(
+            [
+                content_format.value,
+                "version",
+                *(["metadata.labels"] if include_labels else []),
+            ]
+        )
```
Comment (nakamasato) [inline libs/community/langchain_community/document_loaders/confluence.py:345]: `expand` is a comma-separated query parameter. originally hardcoded as `f"{content_format.value},version"`.

I made a variable so we can add more option if necessary in the future as `expand` parameter supports a lot more values.

**Review by efriis — APPROVED** [review]
thanks for the docs screenshot!

**Pull request merged**